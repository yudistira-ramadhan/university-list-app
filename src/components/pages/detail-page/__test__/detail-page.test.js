import React from 'react'
import ReactDOM from 'react-dom'
import DetailPage from '../detail-page'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<DetailPage/>, div)
})

it('renders DetailPage correctly', () => {
  const { getByTestId } = render(<DetailPage />)
  expect(getByTestId('detail-page')).toBeTruthy()
})

it('matches snapshot', () => {
  const tree = renderer.create(<DetailPage />).toJSON()
  expect(tree).toMatchSnapshot()
})