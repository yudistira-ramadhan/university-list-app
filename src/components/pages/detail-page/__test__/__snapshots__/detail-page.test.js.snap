// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="detail-page"
>
  <div
    className="text-center jumbotron"
  >
    <h2
      className="font-weight-bolder"
    >
      University List by Detailed Input
    </h2>
    <hr
      className="my-4"
    />
    <p
      className="lead"
    >
      You can search the University by inputting specific University name and Country.
    </p>
    <div
      className="text-center"
    >
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="bg-info text-light input-group-text"
          >
            Name
          </span>
        </div>
        <input
          className="form-control"
          name="university"
          onChange={[Function]}
          placeholder="Input university name..."
          type="text"
        />
      </div>
      <div
        className="mt-2 input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="bg-info text-light input-group-text"
          >
            Country
          </span>
        </div>
        <input
          className="form-control"
          name="country"
          onChange={[Function]}
          placeholder="Input country name..."
          type="text"
        />
      </div>
      <button
        aria-label={null}
        className="mt-2 w-50 btn btn-outline-primary"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
